core/domain/state_domain.py

```diff
@@ -862,7 +862,9 @@ def validate(self, expected_content_id_list):
                     raise utils.ValidationError(
                         'Expected language_code to be a string, received %s'
                         % language_code)
-                if not utils.is_valid_language_code(language_code):
+                allowed_language_codes = [language['id'] for language in (
+                    constants.SUPPORTED_AUDIO_LANGUAGES)]
```

Comment:
Should we rename constant.SUPPORTED_AUDIO_LANGUAGES to constant.SUPPORTED_ASSETS_LANGUAGES or constant.SUPPORTED_ACCESSIBILITY_LANGUAGES such that it can be used for both voiceovers and translations? (not sure about a generic name)

OR 

Should we change is_valid_language_code to accept SUPPORTED_AUDIO_LANGUAGES?
But this method is used for validating exploration language so I think we cannot modify this method. Also, I'm not sure why we allow translation in Hinglish but exploration cannot be in Hinglish? (Thinking about replacing translated text in exploration player (in future) functionality.)

Reply:
Hi @DubeySandeep, I think we discussed this, and decided that language is complicated :) What did we resolve to do in the short-term?

Reply:
Currently, through the written_translation project, we want to provide a place for translators to write the translation before adding voiceover so I think we can have the same set of languages for Voiceover and translations.

In the long run, we are thinking of showcasing this translated text as the main text for exploration in a different language as well as highlighting text with respect to the voiceover, and this is going to be a very long project  but having written translation for all the language in which we can have voiceover will not be an big issue and can be transformed into our requirement once will finalize the structure for the language.

(Note: I'm relaying with this quick solution as there is an issue in the current translation tab i.e, we need translated text for doing voiceover and translators aren't allowed to add text in Hinglish so they cannot add voice-over (in Hinglish) either.)